testsuite/domain/src/test/java/org/jboss/as/test/integration/domain/events/JmxControlledStateNotificationsTestCase.java

```diff
@@ -82,14 +100,125 @@ public static void tearDownDomain() throws Exception {
+                        "suspended", "stopping",
+                        "starting", "suspended",
+                        "suspended", "normal")
+        );
+    }
+
+    /**
+     * Force transition of a server into restart-required state.
+     */
+    private void forceRestartRequired(ModelControllerClient client, String host, String server)
+            throws UnsuccessfulOperationException, IOException {
+        ModelNode response = client.execute(Operations.createOperation("server-set-restart-required",
+                PathAddress.pathAddress("host", host).append("server", server).toModelNode()));
+        Assert.assertTrue(response.toString(), Operations.isSuccessfulOutcome(response));
+    }
+
+    /**
+     * Reload the whole domain
+     */
+    private void reload(DomainTestSupport testSupport) throws Exception {
+        ModelNode reload = Util
+                .createEmptyOperation("reload", testSupport.getDomainMasterLifecycleUtil().getAddress());
+        testSupport.getDomainMasterLifecycleUtil().executeAwaitConnectionClosed(reload);
+        testSupport.getDomainMasterLifecycleUtil().connect();
+        testSupport.getDomainMasterLifecycleUtil().awaitServers(System.currentTimeMillis());
+    }
+
+    /**
+     * Restart  a particular server in the domain
+     */
+    private void restart(DomainTestSupport testSupport, String host, String server) throws Exception {
+        ModelNode response = testSupport.getDomainMasterLifecycleUtil().getDomainClient()
+                .execute(Operations.createOperation("restart",
+                    PathAddress.pathAddress("host", host).append("server", server).toModelNode()));
+        Assert.assertTrue(response.toString(), Operations.isSuccessfulOutcome(response));
+        testSupport.getDomainMasterLifecycleUtil().awaitServers(System.currentTimeMillis());
+    }
+
+    private List<Pair<String, String>> createListOf(String... transitionPairs) {
+        int i = 0;
+        List<Pair<String, String>> outcome = new ArrayList<>();
+        while (i < transitionPairs.length - 1) {
+            outcome.add(new ImmutablePair<>(transitionPairs[i], transitionPairs[i + 1]));
+            i += 2;
+        }
+        return outcome;
+    }
+
+
+    private void checkFacadeJmxNotifications(List<Pair<String, String>> configurationStateTransitions,
+                                             List<Pair<String, String>> runningStateTransitions)
+            throws IOException, InterruptedException {
         final long end = System.currentTimeMillis() + TimeoutUtil.adjust(20000);
         while (true) {
             try {
-                checkFacadeJmxNotifications();
+                readAndCheckFile(JMX_FACADE_RUNTIME, list -> {
+                    Assert.assertEquals(Arrays.toString(list.toArray(new String[list.size()])),
+                            configurationStateTransitions.size(), list.size());
+                    for (int i = 0; i < configurationStateTransitions.size(); i++) {
```

Comment:
Using i in the error message might make sense, otherwise why use an index  when a simple foreach loop would do the job